### PR TITLE
added filter by name and position to User/find;

### PR DESCRIPTION
**Pull request by vorobevamv**
updated Kernel version, removed BaseFindFilterValidation

### CODE REVIEW

File: src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs [hunk 1]
```diff
@@ -22,71 +19,75 @@ namespace LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers
...
+      IEnumerable<Guid> usersIds = projectUsers.Select(pu => pu.UserId);
 
-      if (projectUsers is null)
+      if (filter.ByPositionId.HasValue)
       {
-        return new();
+        PositionFilteredData data = (await _positionService.GetPositionFilteredDataAsync(
+          new List<Guid>() { filter.ByPositionId.Value }, errors))?.FirstOrDefault();
+
+        usersIds = data is not null ? usersIds.Where(i => data.UsersIds.Contains(i)).ToList() : Enumerable.Empty<Guid>();
```
Comment (VLiamin) [inline src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs:62]: May be "usersIds.Where(data.UsersIds.Contains)"?

File: src/ProjectService/Controllers/UserController.cs [hunk 2]
```diff
@@ -23,11 +24,12 @@ public class UserController : ControllerBase
 
     [HttpGet("find")]
     public async Task<FindResultResponse<UserInfo>> FindAsync(
-      [FromServices] IFindProjectUsersCommand command,
-      [FromQuery] Guid projectId,
-      [FromQuery] FindProjectUsersFilter filter)
+       [FromServices] IFindProjectUsersCommand command,
```
Comment (VLiamin) [inline src/ProjectService/Controllers/UserController.cs:27]: 3 spaces instead of 2

**Review by VLiamin — APPROVED** [review]

File: src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs (hunk 1 above)
Comment (vorobevamv) [reply src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs:62]: done

File: src/ProjectService.Broker/Requests/UserService.cs [hunk 3]
```diff
@@ -36,7 +37,10 @@ public class UserService : IUserService
       _globalCache = globalCache;
     }
 
-    public async Task<(List<UserData> usersData, int totalCount)> GetFilteredUsersAsync(List<Guid> usersIds, FindProjectUsersFilter filter)
+    public async Task<(List<UserData> usersData, int totalCount)> GetFilteredUsersAsync(
+          List<Guid> usersIds,
```
[the remark this answers is not in the document]
Comment (vorobevamv) [reply src/ProjectService.Broker/Requests/UserService.cs:41]: fixed

File: src/ProjectService/Controllers/UserController.cs (hunk 2 above)
Comment (vorobevamv) [reply src/ProjectService/Controllers/UserController.cs:27]: fixed

File: src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs [hunk 4]
```diff
@@ -22,71 +19,75 @@ namespace LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers
...
-            imageSource: ImageSource.User)
+          imagesIds: usersData?.Where(ud => ud.ImageId.HasValue).Select(ud => ud.ImageId.Value).ToList(),
+          imageSource: ImageSource.User,
+          errors,
+          cancellationToken)
         : Task.FromResult<List<ImageInfo>>(default);
 
       Task<List<PositionData>> usersPositionsTask = filter.IncludePositions
-        ? _positionService.GetPositionsAsync(usersIds: filteredUsersData.usersData?.Select(x => x.Id).ToList())
+        ? _positionService.GetPositionsAsync(usersIds: usersData.Select(ud => ud.Id).ToList(), errors, cancellationToken)
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs:76]: userData?.Select...

File: src/ProjectService.Models.Dto/Requests/User/FindProjectUsersFilter.cs [hunk 5]
```diff
@@ -8,13 +9,19 @@ public record FindProjectUsersFilter : BaseFindFilter
...
+    public string FullNameIncludeSubstring { get; set; }
 
     [FromQuery(Name = "includeAvatars")]
     public bool IncludeAvatars { get; set; } = false;
 
     [FromQuery(Name = "includePositions")]
     public bool IncludePositions { get; set; } = false;
+
+    [FromQuery(Name = "byPositionId")]
+    public Guid? ByPositionId { get; set; } = null;
```
Comment (ndoriomedov) [inline src/ProjectService.Models.Dto/Requests/User/FindProjectUsersFilter.cs:25]: PositionId would be better I think

**Review by ndoriomedov — APPROVED** [review]

File: src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs (hunk 4 above)
Comment (vorobevamv) [reply src/ProjectService.Business/Commands/ProjectUsers/FindProjectUsersCommand.cs:76]: oops. thanks. fixed

File: src/ProjectService.Models.Dto/Requests/User/FindProjectUsersFilter.cs (hunk 5 above)
Comment (vorobevamv) [reply src/ProjectService.Models.Dto/Requests/User/FindProjectUsersFilter.cs]: mmm..may be. OK, renamed

**Pull request merged**